dist/codecov.sh

```diff
@@ -0,0 +1,234 @@
+  fi
+}
+write_truthy_args() {
+  if [ "$(eval echo \$$1)" = "true" ] || [ "$(eval echo \$$1)" = "1" ];
+  then
+    echo " -$(lower $1)"
+  fi
+}
+b="\033[0;36m"  # variables/constants
+g="\033[0;32m"  # info/debug
+r="\033[0;31m"  # errors
+x="\033[0m"
+say "     _____          _
+    / ____|        | |
+   | |     ___   __| | ___  ___ _____   __
+   | |    / _ \\ / _\` |/ _ \\/ __/ _ \\ \\ / /
+   | |___| (_) | (_| |  __/ (_| (_) \\ V /
+    \\_____\\___/ \\__,_|\\___|\\___\\___/ \\_/
+                           $r Wrapper-$CC_WRAPPER_VERSION$x
+                                  "
+CC_VERSION="${CC_VERSION:-latest}"
+CC_FAIL_ON_ERROR="${CC_FAIL_ON_ERROR:-false}"
+if [ -n "$CC_BINARY" ];
+then
+  if [ -f "$CC_BINARY" ];
+  then
+    cc_filename=$CC_BINARY
+  else
+    exit_if_error "Could not find binary file $CC_BINARY"
+  fi
+else
+  if [ -n "$CC_OS" ];
+  then
+    say "$g==>$x Overridden OS: $b${CC_OS}$x"
+    export cc_os=${CC_OS}
+  else
+    CC_OS="linux"
+    family=$(uname -s | tr '[:upper:]' '[:lower:]')
+    cc_os="windows"
+    [[ $family == "darwin" ]] && cc_os="macos"
+    [[ $family == "linux" ]] && cc_os="linux"
+    [[ $cc_os == "linux" ]] && \
+      osID=$(grep -e "^ID=" /etc/os-release | cut -c4-)
+    [[ $osID == "alpine" ]] && cc_os="alpine"
+    [[ $(arch) == "aarch64" && $family == "linux" ]] && cc_os+="-arm64"
+    say "$g==>$x Detected $b${cc_os}$x"
+    export cc_os=${cc_os}
+  fi
+  export cc_version=${CC_VERSION}
+  cc_filename="codecov"
+  [[ $cc_os == "windows" ]] && cc_filename+=".exe"
+  export cc_filename=${cc_filename}
+  [[ $cc_os == "macos" ]]  && \
+    ! command -v gpg 2>&1 >/dev/null && \
+    HOMEBREW_NO_AUTO_UPDATE=1 brew install gpg
+  cc_url="https://cli.codecov.io"
+  cc_url="$cc_url/${CC_VERSION}"
+  cc_url="$cc_url/${cc_os}/${cc_filename}"
+  say "$g ->$x Downloading $b${cc_url}$x"
+  curl -Os $cc_url
+  say "$g==>$x Finishing downloading $b${cc_os}:${CC_VERSION}$x"
+  say " "
+fi
+if [ "$CC_SKIP_VALIDATION" = "true" ] || [ -n "$CC_BINARY" ];
+then
+  say "$r==>$x Bypassing validation as requested by user"
+else
+CC_PUBLIC_PGP_KEY=$(curl https://keybase.io/codecovsecurity/pgp_keys.asc)
+  echo "${CC_PUBLIC_PGP_KEY}"  | \
+    gpg --no-default-keyring --import
+  # One-time step
+  say "$g==>$x Verifying GPG signature integrity"
+  sha_url="https://cli.codecov.io"
+  sha_url="${sha_url}/${cc_version}/${cc_os}"
+  sha_url="${sha_url}/${cc_filename}.SHA256SUM"
+  say "$g ->$x Downloading $b${sha_url}$x"
+  say "$g ->$x Downloading $b${sha_url}.sig$x"
+  say " "
+  curl -Os "$sha_url"
+  curl -Os "${sha_url}.sig"
+  if ! gpg --verify "${cc_filename}.SHA256SUM.sig" "${cc_filename}.SHA256SUM";
+  then
+    exit_if_error "Could not verify signature. Please contact Codecov if problem continues"
+  fi
+  if ! (shasum -a 256 -c "${cc_filename}.SHA256SUM" || \
+    sha256sum -c "${cc_filename}.SHA256SUM");
+  then
+    exit_if_error "Could not verify SHASUM. Please contact Codecov if problem continues"
```

Comment:
We might want to tell customers to contact security@ at first sight. I know it's usually harmless, but I think we would prefer the false alarms over a delay of a possible real issue.